core/runtime/src/main/java/io/quarkus/runtime/configuration/ssl/ServerSslConfig.java

```diff
@@ -126,8 +127,9 @@ public SSLContext toSSLContext() throws GeneralSecurityException, IOException {
             if (keyItr.hasNext()) {
                 log.warnf("Ignoring extra content in key file \"%s\"", keyPath);
             }
+            //Entry password needs to match the keystore password
             keyStore.setEntry("default", new KeyStore.PrivateKeyEntry(privateKey, certList.toArray(new X509Certificate[0])),
```

Comment:
The password for individual entries is not connected to the keystore password.  They should be separately configured (and optional as not every keystore implementation will support or even require a password).

Reply:
Furthermore I think we need to have a brief design discussion about how we should store credentials like keystore passwords.

Reply:
> Furthermore I think we need to have a brief design discussion about how we should store credentials like keystore passwords.

I would say that they should only be set in a configuration file if the file itself is not (easily) accessible, like a Kubernetes secret. Otherwise (in my experience) it should be supplied using an environment variable. The latter is supported via Microprofile Config, which picks it up under `QUARKUS_HTTP_SSL_CERTIFICATE_KEY_STORE_PASSWORD`.

Reply:
> The password for individual entries is not connected to the keystore password. They should be separately configured (and optional as not every keystore implementation will support or even require a password).

This may be a strange quirk, but it actually does for the 'default' entry (that's why it was in the original code).  I agree it would be better to specify the entry to lookup and the accompanying password, but that was not the original issue. I'm happy to change it though :-).

Reply:
@dmlloyd I think this PR is an improvement over what we have and I would lean towards getting it merged and potentially create new issues for the future desired improvements. WDYT?

Reply:
@gsmet Sorry, I have been very busy with my day job this last week. I should have some time to work on it next weekend or witha bit of luck in a couple of hours. I'm already working on the integration tests, but it means a bit of an overhaul in the QuarkusUnitTest and accompanying RestAssuredURLManager, TestHTTPResourceManager etc.

Reply:
Sure, if you think it's OK I'll dismiss my review.

Reply:
@dmlloyd @gsmet I added a preliminary test to verify the secure connection works if a keystore is provided. Downside is that I had to change the `ServerSSLConfig` class to first check if the keystore path is referring to a provided resource instead of a physical file somewhere on the file system.